examples/models/object_detection/yolox/pascal_voc/train.py

```diff
@@ -0,0 +1,456 @@
+
+
+def resize_fn(image, gt_boxes, gt_classes):
+    image, image_info = resize_and_crop_image(
+        image, image_size[:2], image_size[:2], 0.8, 1.25
+    )
+    gt_boxes = resize_and_crop_boxes(
+        gt_boxes, image_info[2, :], image_info[1, :], image_info[3, :]
+    )
+    indices = get_non_empty_box_indices(gt_boxes)
+    gt_boxes = tf.gather(gt_boxes, indices)
+    gt_classes = tf.gather(gt_classes, indices)
+    return image, gt_boxes, gt_classes
+
+
+def flip_fn(image, boxes):
+    if tf.random.uniform([], minval=0, maxval=1, dtype=tf.float32) > 0.5:
+        image = tf.image.flip_left_right(image)
+        y1, x1, y2, x2 = tf.split(boxes, num_or_size_splits=4, axis=-1)
+        boxes = tf.concat([y1, 1.0 - x2, y2, 1.0 - x1], axis=-1)
+    return image, boxes
+
+
+def proc_train_fn(bounding_box_format, img_size):
+    def apply(inputs):
+        image = inputs["image"]
+        image = tf.cast(image, tf.float32)
+        gt_boxes = inputs["objects"]["bbox"]
+        image, gt_boxes = flip_fn(image, gt_boxes)
+        gt_boxes = keras_cv.bounding_box.convert_format(
+            gt_boxes,
+            images=image,
+            source="rel_yxyx",
+            target="yxyx",
+        )
+        gt_classes = tf.cast(inputs["objects"]["label"], tf.float32)
+        image, gt_boxes, gt_classes = resize_fn(image, gt_boxes, gt_classes)
+        gt_classes = tf.expand_dims(gt_classes, axis=-1)
+        bounding_boxes = tf.concat([gt_boxes, gt_classes], axis=-1)
+        bounding_boxes = keras_cv.bounding_box.convert_format(
+            bounding_boxes, images=image, source="yxyx", target=bounding_box_format
+        )
+        return {"images": image, "bounding_boxes": bounding_boxes}
+
+    return apply
+
+
+train_ds = train_ds.map(
+    proc_train_fn("xywh", image_size), num_parallel_calls=tf.data.AUTOTUNE
+)
+train_ds = train_ds.apply(
+    tf.data.experimental.dense_to_ragged_batch(BATCH_SIZE, drop_remainder=True)
+)
+eval_ds = eval_ds.map(
+    proc_train_fn(bounding_box_format="xywh", img_size=image_size),
+    num_parallel_calls=tf.data.AUTOTUNE,
+)
+eval_ds = eval_ds.apply(
+    tf.data.experimental.dense_to_ragged_batch(BATCH_SIZE, drop_remainder=True)
+)
+
+"""
+Looks like everything is structured as expected.  Now we can move on to constructing our
+data augmentation pipeline.
+"""
+train_ds = train_ds.prefetch(2)
+train_ds = train_ds.shuffle(BATCH_SIZE)
+eval_ds = eval_ds.prefetch(2)
+
+
+def unpackage_dict(inputs):
+    return inputs["images"], inputs["bounding_boxes"]
+
+
+train_ds = train_ds.map(unpackage_dict, num_parallel_calls=tf.data.AUTOTUNE)
+eval_ds = eval_ds.map(unpackage_dict, num_parallel_calls=tf.data.AUTOTUNE)
+
+"""
+Our data pipeline is now complete.  We can now move on to model creation and training.
+"""
+
+"""
+## Model creation
+
+We'll use the KerasCV API to construct a YoloX_tiny model.  In this tutorial we use
+a pretrained ResNet50 backbone using weights.  In order to perform fine-tuning, we
```

Comment:
but below it uses cspdarknet, not resnet?

Reply:
Oops, small pasting error. Will fix this.